File: examples/blobs/bucket-positional-args.ts
import type { JsBaoClient } from "js-bao-wss-client";

// The bucket API is flat: there is NO bucket() context object. The bucket
// key (or id) is always the first positional argument to each method.
export async function bucketPositionalArgs(
  client: JsBaoClient,
  blobId: string,
  filename: string,
  contentType: string,
  data: Uint8Array,
) {
  // #region example
  // The bucket key/ID is a positional arg — there is no bucket() context object.
  await client.blobBuckets.upload("avatars", { filename, contentType, data });
  await client.blobBuckets.getSignedUrl("avatars", blobId, 3600);
  // #endregion example
}

File: examples/blobs/bucket-positional-args.swift
import Foundation
import JsBaoClient

// The bucket API is flat: there is no bucket() context object. The bucket
// key (or id) is always the leading argument to each method.
func bucketPositionalArgs(
  client: JsBaoClient,
  blobId: String,
  filename: String,
  contentType: String,
  data: Data
) async throws {
  // #region example
  // The bucket key/ID is a positional arg — there is no bucket() context object.
  try await client.blobBuckets.upload(
    bucketIdOrKey: "avatars", data: data, filename: filename, contentType: contentType
  )
  try await client.blobBuckets.getSignedUrl(
    bucketIdOrKey: "avatars", blobId: blobId, expiresInSeconds: 3600
  )
  // #endregion example
}
